client/src/debug_config_provider.ts

```diff
@@ -30,8 +31,18 @@ export class DenoDebugConfigurationProvider
     return args;
   }
 
-  constructor(getSettings: () => Settings) {
-    this.#getSettings = getSettings;
+  #getInspectArg() {
+    const version = this.#extensionContext.serverInfo?.version;
+
+    if (version && semver.valid(version) && semver.satisfies(version, ">=1.29.0")) {
+      return "--inspect-wait"
+    } else {
+      return "--inspect";
```

Comment:
Now that I'm thinking about it... we should probably default to `--inspect-brk` in older versions of Deno, otherwise some breakpoints might not get registered and hit. @Leokuma could you change that?

Reply:
Sure! New PR incoming.

Reply:
#779